### PR TITLE
PRIME-2393 all nurses have pharmanet

### PR DESCRIPTION
**Pull request by yousiefc**
- Changed `PrescriberIdType` for Nurses to `Mandatory`

### CODE REVIEW

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public class LicenseDetailConfiguration : SeededTable<LicenseDetail>
         public static readonly DateTime LicenceClassUpdateAug5th2022 = new(2022, 8, 5, 8, 0, 0, DateTimeKind.Utc);
         public static readonly DateTime EnableNonPrescriberPrefixAug10th2022 = new(2022, 8, 10, 8, 0, 0, DateTimeKind.Utc);
         public static readonly DateTime SetPractIdToMandatorySept6th2022 = new(2022, 9, 6, 8, 0, 0, DateTimeKind.Utc);
+        public static readonly DateTime SetPractIdToMandatoryFeb1st2023 = new(2023, 2, 1, 8, 0, 0, DateTimeKind.Utc);
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs:18]: Can you make this Feb 2 as that date is the "special" date?

Comment (yousiefc) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs:18]: Yes. Just to confirm, you mean to change the name to `SetPractIdToMandatoryFeb2nd2023`?

Comment (neophyte57) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs:18]: And `new(2023, 2, 2, 8, 0, 0, DateTimeKind.Utc);` just to be clear

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs [hunk 2]
```diff
@@ -248,6 +249,24 @@ public override IEnumerable<LicenseDetail> SeedData
                     new LicenseDetail { Id = 200, LicenseCode = 60, Prefix = "98", NamedInImReg = true, LicensedToProvideCare = true, Manual = false, Validate = true, PrescriberIdType = PrescriberIdType.Mandatory, AllowRequestRemoteAccess = true, EffectiveDate = SetPractIdToMandatorySept6th2022, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
                     new LicenseDetail { Id = 201, LicenseCode = 61, Prefix = "98", NamedInImReg = false, LicensedToProvideCare = true, Manual = false, Validate = true, PrescriberIdType = PrescriberIdType.Mandatory, AllowRequestRemoteAccess = true, EffectiveDate = SetPractIdToMandatorySept6th2022, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
                     new LicenseDetail { Id = 202, LicenseCode = 62, Prefix = "98", NamedInImReg = true, LicensedToProvideCare = true, Manual = false, Validate = true, PrescriberIdType = PrescriberIdType.Mandatory, AllowRequestRemoteAccess = true, EffectiveDate = SetPractIdToMandatorySept6th2022, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
+                 
+                    // Feb 1st 2023
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs:253]: Change this comment as well please

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs (hunk 1 above)
Comment (yousiefc) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs:18]: You got it

**Comment by neophyte57** [comment]
There is something not quite right as `prime-dotnet-webapi/Migrations/20230207222353_AllNursesHavePharmanetId.cs` has changes to `AgreementVersion` which doesn't make sense.  It may be because of the state of `ApiDbContextModelSnapshot.cs` before you executed `dotnet ef migrations add`

**Comment by yousiefc** [comment]
> There is something not quite right as `prime-dotnet-webapi/Migrations/20230207222353_AllNursesHavePharmanetId.cs` has changes to `AgreementVersion` which doesn't make sense.  It may be because of the state of `ApiDbContextModelSnapshot.cs` before you executed `dotnet ef migrations add`
> 

Oh weird. Would it be best then to rollback to the previous snapshot and try migrating again?

File: prime-dotnet-webapi/Migrations/20230208234852_AllNursesHavePharmanetId.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+namespace Prime.Migrations
+{
+    public partial class AllNursesHavePharmanetId : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropCheckConstraint(
+                name: "CHK_Agreement_EitherPartyOrEnrollee",
+                table: "Agreement");
+
```
Comment (neophyte57) [inline prime-dotnet-webapi/Migrations/20230208234852_AllNursesHavePharmanetId.cs:13]: I think you also have to add `prime-dotnet-webapi/Configuration/Database/Unseeded Tables/AgreementConfiguration.cs` to `.gitattributes`

**Review by neophyte57 — APPROVED** [review]
Woo-hoo 🎉 , LGTM

**Pull request merged**